webview-ui/src/CreateFleet/CreateFleetInput.tsx

```diff
@@ -0,0 +1,155 @@
+import { EventDef } from "./helpers/state";
+import { EventHandlers } from "../utilities/state";
+import { just, Maybe, nothing } from "../utilities/maybe";
+
+type ChangeEvent = Event | FormEvent<HTMLElement>;
+
+interface CreateFleetInputProps {
+    locations: string[];
+    resourceGroups: ResourceGroup[];
+    eventHandlers: EventHandlers<EventDef>;
+    vscode: MessageSink<ToVsCodeMsgDef>;
+}
+
+export function CreateFleetInput(props: CreateFleetInputProps) {
+    const [existingResourceGroup, setExistingResourceGroup] = useState<Validatable<ResourceGroup | null>>(unset());
+    const [fleetName, setFleetName] = useState<Validatable<string>>(unset());
+    // if show new resources group
+    // new resource group name
+    // hub cluster mode selection button
+    const [selectedIndex, setSelectedIndex] = useState<number>(0); // resource group
+    const [location, setLocation] = useState<Validatable<string>>(unset());
+
+    const allResourcesGroups = props.resourceGroups;
+
+    function handleValidationAndIndex(e: ChangeEvent) {
+        handleExistingResourceGroupChange(e);
+        const ele = e.currentTarget as HTMLSelectElement;
+        setSelectedIndex(ele.selectedIndex);
+    }
+
+    function handleExistingResourceGroupChange(e: ChangeEvent) {
+        const elem = e.currentTarget as HTMLSelectElement;
+        const resourceGroup = elem.selectedIndex <= 0 ? null : allResourcesGroups[elem.selectedIndex - 1];
+        const validatable = resourceGroup ? valid(resourceGroup) : invalid(null, "Resource Group is required.");
+        setExistingResourceGroup(validatable);
+    }
+
+    function handleFleetNameChange(e: ChangeEvent) {
+        const name = (e.currentTarget as HTMLInputElement).value;
+        const validated = getValidatedName(name);
+        setFleetName(validated);
+    }
+
+    function getValidatedName(name: string): Validatable<string> {
+        if (!name) return invalid(name, "Fleet name must be at least 1 character long.");
+        if (name.length > 63) return invalid(name, "Fleet name must be at most 63 characters long.");
+        if (!/^[a-zA-Z0-9][a-zA-Z0-9_-]*[a-zA-Z0-9]$/.test(name)) {
+            return invalid(
+                name,
+                "The only allowed characters are letters, numbers, dashes, and underscore. The first and last character must be a letter or a number.",
+            );
+        }
+
+        return valid(name);
+    }
+
+    function validate(): Maybe<CreateFleetParams> {
+        if (!isValid(location)) return nothing();
+        let resourceGroupName: string;
+        let isNewResourceGroup: boolean;
+        if (isValid(existingResourceGroup) && existingResourceGroup.value !== null) {
+            resourceGroupName = existingResourceGroup.value.name;
+            isNewResourceGroup = false;
+            // } else if (newResourceGroupName) {
+            //     resourceGroupName = newResourceGroupName;
+            //     isNewResourceGroup = true;
```

Comment:
can we delete _all_ code related to "is new resource group" and just require the resource group to be pre-created at first? we can think about this create RG flow later if necessary